### PR TITLE
Setup Search module to interact with the Search Rest Client API.

### PR DESCRIPTION
**Pull request by PenghaiZhang**
#1306 

##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change
The Search module provides a function that interacts with REST Client API to do a search.

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchModule.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as OEQ from "@openequella/rest-api-client";
+import { Config } from "../config";
+
+const API_BASE_URL = `${Config.baseUrl}api`;
+
+export const searchItems = (
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchModule.ts:23]: Could we have some JSDoc please?

File: Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchModule.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as OEQ from "@openequella/rest-api-client";
+import { Config } from "../config";
+
+const API_BASE_URL = `${Config.baseUrl}api`;
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchModule.ts:21]: I swear I've seen this constant just like this in other modules. Is it worth centralizing it somewhere?

Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchModule.ts:21]: Maybe we can move this to `tsrc/api/General.ts` ?

Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchModule.ts:21]: As discussed, Config is more ideal.

**Pull request merged**